virttest/utils_libvirt/libvirt_network.py

```diff
@@ -106,43 +108,45 @@ def _check_socket():
     else:
         server_ip = params.get("server_ip", params.get("remote_ip"))
 
-    cmd = "netstat -tunap|grep %s" % port_to_check
-    if check_local:
-        cmdRes = process.run(cmd, shell=True)
-    else:
-        cmdRes = remote.run_remote_cmd(cmd, params)
-
-    if expected_network_conn_num:
-        pat_str = r'.*%s:.*ESTABLISHED.*qemu-kvm.*' % server_ip
-        findall = re.findall(pat_str, cmdRes.stdout_text.strip())
-        if len(findall) != int(expected_network_conn_num):
-            raise exceptions.TestFail("Failed to check network connection between "
-                                      "src qemu and target qemu: {}".format(findall))
-        else:
-            LOG.debug("Network connection number: %s", len(findall))
-        if int(expected_network_conn_num) == 0:
-            return None
-
-    if port_to_check != '4915':
-        pat_str = r'.*%s:%s.*ESTABLISHED.*qemu-kvm.*' % (server_ip,
-                                                         port_to_check)
-        search = re.search(pat_str, cmdRes.stdout_text.strip())
-        if not search:
-            raise exceptions.TestFail("Pattern '%s' is not matched in "
-                                      "'%s'" % (pat_str,
-                                                cmdRes.stdout_text.strip()))
+    if check_port_or_network_conn_num:
+        cmd = "netstat -tunap|grep %s" % port_to_check
+        if check_local:
+            cmdRes = process.run(cmd, shell=True)
         else:
-            return port_to_check
-    else:
-        pat_str = r'.*%s:(\d*).*ESTABLISHED.*qemu-kvm.*' % server_ip
-        search = re.search(pat_str, cmdRes.stdout_text.strip())
-        if search:
-            LOG.debug("Get the port used:%s", search.group(1))
-            return search.group(1)
+            cmdRes = remote.run_remote_cmd(cmd, params)
+
+        if expected_network_conn_num:
+            pat_str = r'.*%s:.*ESTABLISHED.*%s.*' % (server_ip, service_to_check)
+            findall = re.findall(pat_str, cmdRes.stdout_text.strip())
+            if len(findall) != int(expected_network_conn_num):
+                raise exceptions.TestFail("Failed to check network connection between "
+                                          "src qemu and target qemu: {}".format(findall))
+            else:
+                LOG.debug("Network connection number: %s", len(findall))
+            if int(expected_network_conn_num) == 0:
+                return None
+
+        if port_to_check != '4915':
```

Comment:
hi liping , just confirm ,does this port "4915" mean some ,do we need to add it into cfg?

Reply:
I think we don't need to add it into cfg. Be default, the migration ports used by target are in range 49152~49215. If 'port_to_check' is not '4915', we will check 'port_to_check'. Otherwise, we will check migration ports.